test/unit/os/fs_spec.lua

```diff
@@ -501,6 +511,59 @@ describe('fs function', function()
       end)
     end)
 
+    describe('os_mkdir_recurse', function()
+      it('returns zero when given an already existing directory', function()
+        local mode = ffi.C.kS_IRUSR + ffi.C.kS_IWUSR + ffi.C.kS_IXUSR
+        local ret, failed_str = os_mkdir_recurse('unit-test-directory', mode)
+        eq(0, ret)
+        eq(nil, failed_str)
+      end)
+
+      it('fails to create a directory where there is a file', function()
+        local mode = ffi.C.kS_IRUSR + ffi.C.kS_IWUSR + ffi.C.kS_IXUSR
+        local ret, failed_str = os_mkdir_recurse(
+            'unit-test-directory/test.file', mode)
+        neq(0, ret)
+        eq('unit-test-directory/test.file', failed_str)
+      end)
+
+      it('fails to create a directory where there is a file in path', function()
+        local mode = ffi.C.kS_IRUSR + ffi.C.kS_IWUSR + ffi.C.kS_IXUSR
+        local ret, failed_str = os_mkdir_recurse(
+            'unit-test-directory/test.file/test', mode)
+        neq(0, ret)
+        eq('unit-test-directory/test.file', failed_str)
+      end)
+
+      it('succeeds to create a directory', function()
+        local mode = ffi.C.kS_IRUSR + ffi.C.kS_IWUSR + ffi.C.kS_IXUSR
```

Comment:
This is consistent with the surrounding code. I do not want to change the surroundings in the unrelated PR.